In [1]:
import requests
import json

In [4]:
response = requests.get(r'http://localhost:5000/api/events/1711485879.618846-sdaj55').text
response = json.loads(response)
response['end_time'] is None

False

In [47]:
for resp in response:
    print(resp['retain_indefinitely'])

True
True
False
False
False
False
False
False
False
False
False
False
False
False
False
False


In [28]:
response[0]['end_time'] - response[0]['start_time']

123.20494103431702

In [1]:
import cv2
import tensorflow as tf
import numpy as np

2024-03-13 01:26:25.788081: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-13 01:26:26.708662: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT


In [3]:
image = cv2.imread('photo.jpg')
image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)

# Ресайз изображения
image = cv2.resize(image, (320, 320), interpolation=cv2.INTER_LINEAR)

# Скалирование изображения
image = image / 255.0
input_tensor = image[np.newaxis, :, :, :].astype(np.float32)

In [4]:
session = tf.lite.Interpreter(model_path='config/yolo_float16.tflite')
session.allocate_tensors()
session.set_tensor(0, input_tensor)

In [6]:
session = tf.lite.Interpreter(model_path='./cpu_model.tflite')

In [7]:
session.get_input_details()

[{'name': 'normalized_input_image_tensor',
  'index': 328,
  'shape': array([  1, 320, 320,   3], dtype=int32),
  'shape_signature': array([  1, 320, 320,   3], dtype=int32),
  'dtype': numpy.uint8,
  'quantization': (0.0078125, 128),
  'quantization_parameters': {'scales': array([0.0078125], dtype=float32),
   'zero_points': array([128], dtype=int32),
   'quantized_dimension': 0},
  'sparsity_parameters': {}}]

In [10]:
session.get_output_details()

[{'name': 'TFLite_Detection_PostProcess',
  'index': 320,
  'shape': array([], dtype=int32),
  'shape_signature': array([], dtype=int32),
  'dtype': numpy.float32,
  'quantization': (0.0, 0),
  'quantization_parameters': {'scales': array([], dtype=float32),
   'zero_points': array([], dtype=int32),
   'quantized_dimension': 0},
  'sparsity_parameters': {}},
 {'name': 'TFLite_Detection_PostProcess:1',
  'index': 321,
  'shape': array([], dtype=int32),
  'shape_signature': array([], dtype=int32),
  'dtype': numpy.float32,
  'quantization': (0.0, 0),
  'quantization_parameters': {'scales': array([], dtype=float32),
   'zero_points': array([], dtype=int32),
   'quantized_dimension': 0},
  'sparsity_parameters': {}},
 {'name': 'TFLite_Detection_PostProcess:2',
  'index': 322,
  'shape': array([], dtype=int32),
  'shape_signature': array([], dtype=int32),
  'dtype': numpy.float32,
  'quantization': (0.0, 0),
  'quantization_parameters': {'scales': array([], dtype=float32),
   'zero_points': 

In [8]:
session.invoke()
index = session.get_output_details()[0]['index']
outputs = session.get_tensor(index)

In [19]:
outputs.shape

(1, 84, 2100)

In [18]:
outputs[0][0].shape

(2100,)

In [19]:
cv2.imshow('test', outputs)

error: OpenCV(4.8.0) /io/opencv/modules/core/src/array.cpp:2494: error: (-206:Bad flag (parameter or structure field)) Unrecognized or unsupported array type in function 'cvGetMat'


In [20]:
cv2.destroyAllWindows()

In [1]:
import frigate

In [3]:
from frigate.util.image import (
    SharedMemoryFrameManager,
    area,
    calculate_region,
    draw_box_with_label,
    draw_timestamp,
    is_label_printable,
)

ModuleNotFoundError: No module named 'frigate.util'

In [27]:
temp = [{'id': '1710450830.682283-ipn4mn', 'camera': 'test', 'frame_time': 1710450830.682283,
  'snapshot': None, 'label': 'person', 'sub_label': None, 'top_score': 0.0, 'false_positive': True,
  'start_time': 1710450830.682283, 'end_time': None, 'score': 0.7989730834960938, 'box': (443, 65, 502, 267),
  'area': 11918, 'ratio': 0.29207920792079206, 'region': (1, 0, 641, 640), 'stationary': False,
  'motionless_count': 0, 'position_changes': 0, 'current_zones': [], 'entered_zones': [], 'has_clip': False,
  'has_snapshot': False, 'attributes': "defaultdict(<class 'float'>, {})", 'current_attributes': []},
  {'id': '1710450830.682283-mpl3iu', 'camera': 'test', 'frame_time': 1710450830.682283, 'snapshot': None,
   'label': 'car', 'sub_label': None, 'top_score': 0.0, 'false_positive': True,
   'start_time': 1710450830.682283, 'end_time': None, 'score': 0.9315774440765381, 'box': (76, 64, 382, 204),
   'area': 42840, 'ratio': 2.1857142857142855, 'region': (1, 0, 641, 640), 'stationary': False,
   'motionless_count': 0, 'position_changes': 0, 'current_zones': [], 'entered_zones': [], 'has_clip': False,
   'has_snapshot': False, 'attributes': "defaultdict(<class 'float'>, {})", 'current_attributes': []},
   {'id': '1710450830.682283-fg6acu', 'camera': 'test', 'frame_time': 1710450830.682283, 'snapshot': None,
    'label': 'person', 'sub_label': None, 'top_score': 0.0, 'false_positive': True, 'start_time': 1710450830.682283,
    'end_time': None, 'score': 0.8512085676193237, 'box': (505, 65, 568, 276), 'area': 13293, 'ratio': 0.2985781990521327,
    'region': (1, 0, 641, 640), 'stationary': False, 'motionless_count': 0, 'position_changes': 0, 'current_zones': [],
    'entered_zones': [], 'has_clip': False, 'has_snapshot': False, 'attributes': "ConnectionError",
    'current_attributes': []}]
temp

[{'id': '1710450830.682283-ipn4mn',
  'camera': 'test',
  'frame_time': 1710450830.682283,
  'snapshot': None,
  'label': 'person',
  'sub_label': None,
  'top_score': 0.0,
  'false_positive': True,
  'start_time': 1710450830.682283,
  'end_time': None,
  'score': 0.7989730834960938,
  'box': (443, 65, 502, 267),
  'area': 11918,
  'ratio': 0.29207920792079206,
  'region': (1, 0, 641, 640),
  'stationary': False,
  'motionless_count': 0,
  'position_changes': 0,
  'current_zones': [],
  'entered_zones': [],
  'has_clip': False,
  'has_snapshot': False,
  'attributes': "defaultdict(<class 'float'>, {})",
  'current_attributes': []},
 {'id': '1710450830.682283-mpl3iu',
  'camera': 'test',
  'frame_time': 1710450830.682283,
  'snapshot': None,
  'label': 'car',
  'sub_label': None,
  'top_score': 0.0,
  'false_positive': True,
  'start_time': 1710450830.682283,
  'end_time': None,
  'score': 0.9315774440765381,
  'box': (76, 64, 382, 204),
  'area': 42840,
  'ratio': 2.1857142857142855,
 

In [30]:
temp[0]['box']

(443, 65, 502, 267)

In [5]:
%cd txts/
!pwd

/home/pc/frigate_test/txts
/home/pc/frigate_test/txts


In [81]:
file_names = !ls
for file_name in file_names:
    newlines = []
    with open(file_name) as file:
        lines = file.readlines()
        # print(lines)
        for line in lines:
            line = line.replace('0', '5', 1) if line[0] == '0' else line.replace('1', '3', 1)
            newlines.append(line)
        # print(f'{newlines}\n')
    with open(file_name, 'w') as file:
        file.writelines(newlines)

In [6]:
ids = ['id0', 'id0', 'id1']
print(ids)
ids.remove('id0')
print(ids)

['id0', 'id0', 'id1']
['id0', 'id1']


In [7]:
import numpy as np

In [12]:
np.max((30, -5))


30

In [33]:
max_detected_speed = 35
speed = 40

max_detected_speed = speed if speed > max_detected_speed else max_detected_speed
max_detected_speed

40

In [3]:
import subprocess

subprocess.call(['python3', 's1.py', '123'])

s1: 123


0

In [38]:
processes

[<Process name='Process-43' pid=100632 parent=90934 stopped exitcode=0>,
 <Process name='Process-44' pid=100634 parent=90934 stopped exitcode=0>]

In [37]:
import multiprocessing
import time

def run_script(sid, camera):
    subprocess.call(['python3', f'{sid}.py', f'{camera}'])
    pass

processes = []

print('before cycle')

for i in range(1, 3):
    process = multiprocessing.Process(target=run_script, args=[f's{i}', 'test'])
    process.start()
    processes.append(process)

print('after cycle')


for i, p in enumerate(processes):
    if p.is_alive():
        continue
    else:
        # Удаляем процесс из списка, если он завершил свою работу
        del processes[i]
        break


before cycle
after cycle


s1: test
s2: test


In [5]:
import os

if not os.path.isdir('storage/speed_estimation/12/'):
    os.mkdir('storage/speed_estimation/12/')

In [ ]:
import argparse
import time

time.sleep(2)
parser = argparse.ArgumentParser()
parser.add_argument('arg')
args = parser.parse_args()
arg = args.arg
print(f's1: {arg}')

In [1]:
import requests

API_URL = 'http://localhost:5000/api/'


def set_retain_to_true(id):
    """
    Sets retain to true for the event with matches id
    """
    url = f'{API_URL}events/{id}/retain'
    # set star
    response = requests.post(url=url)
    # remove star
    # response = requests.delete(url=url)
    try:
        print(response.json())
    except:
        pass



In [7]:
set_retain_to_true('1712242738.560129-zb1hyy')

{'message': 'Event 1712242738.560129-zb1hyy retained', 'success': True}


In [ ]:
'1712242738.560129-lywhms'
'1712242738.560129-zb1hyy'

In [6]:
API_URL = 'http://localhost:5000/api/'
url = f'{API_URL}api/events/{id}/retain'
url

'http://localhost:5000/api/api/events/<built-in function id>/retain'

In [33]:
import json
import numpy as np
with open('speed_estimation/transform_points.json') as file:
    transform_points = json.loads(file.read())
transform_points

{'test': {'TARGET_WIDTH': 25,
  'TARGET_HEIGHT': 250,
  'SOURCE': [[1252, 787], [2298, 803], [5039, 2159], [-550, 2159]]}}

In [28]:
SOURCE = transform_points['test']['SOURCE']

TARGET_WIDTH = np.array(transform_points['test']['TARGET_WIDTH'])
TARGET_HEIGHT = np.array(transform_points['test']['TARGET_HEIGHT'])

TARGET = np.array([
    [0, 0],
    [TARGET_WIDTH - 1, 0],
    [TARGET_WIDTH - 1, TARGET_HEIGHT - 1],
    [0, TARGET_HEIGHT - 1],
])
TARGET

array([[  0,   0],
       [ 24,   0],
       [ 24, 249],
       [  0, 249]])

### docker compose logs -f 


```json
{
    "имя_камеры": {<br/>
        "TARGET_WIDTH" : ширина,<br/>
        "TARGET_HEIGHT" : высота,<br/>
        "SOURCE" : [<br/>
            [1252, 787],<br/>
            [2298, 803],<br/>
            [5039, 2159],<br/>
            [-550, 2159]<br/>
        ]<br/>
    }
}

In [51]:
import numpy as np

bounding_boxes = np.array([[1, 2, 3, 4], [5, 6, 7, 8], [11, 22, 33, 44], [44, 55, 66, 77]])
ids = np.array([1, 2, 1, 2])

print(bounding_boxes)
bounding_boxes = bounding_boxes[ids == 1]
print(f'\n{bounding_boxes}')

[[ 1  2  3  4]
 [ 5  6  7  8]
 [11 22 33 44]
 [44 55 66 77]]

[[ 1  2  3  4]
 [11 22 33 44]]


In [1]:
def sort_rectangle_points(points):
    # Убедимся, что мы получили именно 4 точки
    if len(points) != 4:
        raise ValueError("Необходимо предоставить ровно 4 точки.")

    # Сначала находим верхнюю левую точку
    top_left = min(points, key=lambda p: (p[0], p[1]))
    points.remove(top_left)

    # Теперь у нас остались 3 точки. Определим их
    # По часовой стрелке от верхней левой точки
    # Относительно верхней левой точки
    def angle_from_top_left(point):
        # Считаем угол относительно линии, проведенной через верхнюю левую точку
        x, y = point[0] - top_left[0], point[1] - top_left[1]
        return (y, -x)  # Поменяли знак на x, так как в системе координат y вниз

    # Сортируем оставшиеся точки по углу
    sorted_points = sorted(points, key=angle_from_top_left)

    # Возвращаем список, начинающийся с верхней левой точки
    return [top_left] + sorted_points

In [1]:
import requests
import json
import yaml
import numpy as np

API_URL = 'http://192.168.16.40:5000/api/'
camera_name = 'test'

url = f'{API_URL}config'
response = requests.get(url=url).text
response = json.loads(response)

source = response["cameras"][camera_name]['zones']['zone_0']['coordinates']

source = [int(number) for number in source.split(',')]
source = np.array(source).reshape(-1, 2).tolist()
source

[[285, 979], [717, 802], [134, 156], [65, 177]]

In [11]:
source = [
    [72, 172],
    [115, 153],
    [712, 800],
    [295, 978]
]

In [10]:
import cv2

image = np.zeros((1080,1920,3), np.uint8)

# sorted = sort_rectangle_points(source)
sorted = [[65, 177], [134, 156], [717, 802], [285, 979]]
for id, point in enumerate(sorted):
    cv2.putText(image, str(id), point, cv2.FONT_HERSHEY_COMPLEX, 1, (255, 255, 255))

cv2.imshow('image', image)
while(True):
    if cv2.waitKey(1) & 0xFF == ord('q'):
        cv2.destroyAllWindows()
        break


In [6]:
import requests
import json
import yaml
import numpy as np

def sort_rectangle_points(points):
    # Убедимся, что мы получили именно 4 точки
    if len(points) != 4:
        raise ValueError("Необходимо предоставить ровно 4 точки.")

    # Сначала находим верхнюю левую точку
    top_left = min(points, key=lambda p: (p[0], p[1]))
    points.remove(top_left)

    # Теперь у нас остались 3 точки. Определим их
    # По часовой стрелке от верхней левой точки
    # Относительно верхней левой точки
    def angle_from_top_left(point):
        # Считаем угол относительно линии, проведенной через верхнюю левую точку
        x, y = point[0] - top_left[0], point[1] - top_left[1]
        return (y, -x)  # Поменяли знак на x, так как в системе координат y вниз

    # Сортируем оставшиеся точки по углу
    sorted_points = sorted(points, key=angle_from_top_left)

    # Возвращаем список, начинающийся с верхней левой точки
    return [top_left] + sorted_points

def get_transform_points_from_config(camera: str):
    """
    Get transform points for camera

    :camera: str - camera name
    :return: np.array(int) - transform points for camera
    """
    
    url = f'http://192.168.16.40:5000/api/config'
    response = requests.get(url=url).text
    response = json.loads(response)
    source = response["cameras"][camera]['zones']['zone_0']['coordinates']
    print(source)
    source = [int(number) for number in source.split(',')]
    source = np.array(source).reshape(-1, 2).tolist()
    source = sort_rectangle_points(source)

    with open('mqtt/speed_estimation/transform_points.json') as file:
        transform_points = json.loads(file.read())
    target_width = transform_points[camera]['TARGET_WIDTH'] * 10
    target_height = transform_points[camera]['TARGET_HEIGHT'] * 10
    target = np.array([
        [0, 0],
        [target_width - 1, 0],
        [target_width - 1, target_height - 1],
        [0, target_height - 1],
    ])
    
    return source, target

get_transform_points_from_config('Reg02_cam13')

1743,187,1852,227,377,936,137,588


([[137, 588], [1743, 187], [1852, 227], [377, 936]],
 array([[  0.,   0.],
        [699.,   0.],
        [699.,  55.],
        [  0.,  55.]]))

In [7]:
get_transform_points_from_config('Reg13_cam19')

297,979,719,799,122,156,74,168


([[74, 168], [122, 156], [719, 799], [297, 979]],
 array([[  0.,   0.],
        [ 28.,   0.],
        [ 28., 539.],
        [  0., 539.]]))

In [1]:
points = [
    [0, 0],
    [347, 0],
    [341, 480],
    [0, 480]
    ]
points

[[0, 0], [347, 0], [341, 480], [0, 480]]

In [ ]:
/opt/frigate/web/index.html
/BASE_PATH/assets/favicon-92a9ac8b.ico
<link rel="mask-icon" href="/BASE_PATH/assets/favicon-1264f127.svg" color="#3b82f7" />
<link rel="icon" type="image/svg+xml" href="/BASE_PATH/assets/favicon-1264f127.svg" />

In [2]:
import json
def get_permitted_speed(camera: str):
    """
    Get permitted speed for camera
    :camera: str - camera name

    :return: int - permitted speed for camera
    """
    try:
        with open('mqtt/speed_estimation/transform_points.json') as file:
            transform_points = json.loads(file.read())
        permitted_speed = transform_points[camera]['PERMITTED_SPEED']
    except:
        print(f'No permitted speed for camera \'{camera}\'')
        permitted_speed = None

    return permitted_speed

get_permitted_speed('test2')

No permitted speed for camera 'test2'


In [4]:
from datetime import datetime
import time

start_time = datetime.now()
print(start_time)
time.sleep(3)
end_time = datetime.now()
print(end_time)
int((end_time-start_time).total_seconds())

2024-09-24 18:02:18.841840
2024-09-24 18:02:21.842561


3

In [2]:
from datetime import datetime
camera = 'Reg13_cam19'

start_time = datetime.now()
directory = '/storage/' + start_time.strftime(r'%d.%m.%Y/')
camera_name = camera.lower().replace('reg', 'r').replace('cam', 'c')
# if not os.path.isdir(directory):
#     os.mkdir(directory)
filename = directory + camera_name + \
    start_time.strftime(r'_%H.%M.%S') + '.mp4'
filename

'/storage/27.09.2024/r13_c19_15.44.26.mp4'

Сделать видео с каждой из 6 камер
погрузчики всех цветов
погрузчики с разных ракурсов
погрузчики с катушками на вилах и без
разная освещенность, погода, время суток
желательно, всевозможные комбинации

### Clip download

##### Загрузка клипов

In [ ]:
import requests
import urllib.request as request
import os
import time as system_time
import json
import yaml
import numpy as np


url = 'http://192.168.16.40:5000/api/events'
response = requests.get(url=url).text
response = json.loads(response)
len(response)

for event in response:
    event_id = event['id']
    filename = f'/home/pc/Downloads/clips/{event_id}.mp4'
    try:
        url = f'http://192.168.16.40:5000/api/events/{event_id}/clip.mp4'
        request.urlretrieve(url, filename)
        result = True
        print('Clip downloaded')
    except:
        result = False
        print('Clip download error')

##### Разбивка на кадры
##### Cтруктура:<br>
directory <br>
&nbsp;&nbsp;&nbsp;&nbsp;|_ videos <- тута видевы клипов <br>
&nbsp;&nbsp;&nbsp;&nbsp;|_ frames <- тута будут фреймы (создать папку предварительно)

In [ ]:
import cv2

directory = '/home/pc/Downloads/clips_october'

def stream(id, filename):
    address = f'{directory}/videos/{filename}'
    cv2.namedWindow('stream', cv2.WINDOW_NORMAL)
    cap = cv2.VideoCapture(address)
    # fourcc = cv2.VideoWriter_fourcc(*'mp4v')
    # width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
    # height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
    # fps = int(cap.get(cv2.CAP_PROP_FPS))
    counter = 0
    while cap.isOpened():
        # Кадр с камеры
        ret, frame = cap.read()
        if not ret:
            break
        frame_directory = f'{directory}/frames/'
        if counter % 20 == 0:
            cv2.imwrite(f'{frame_directory}#{id}frame{counter}.png', frame)
        cv2.imshow('stream', frame)
        if cv2.waitKey(1) & 0xFF == ord('q'):
            cv2.destroyAllWindows()
            cap.release()
            return
        counter += 1
        
            

    cv2.destroyAllWindows()
    cap.release()

# stream(r'rtsp://localhost:8554/test')
# stream(r'rtsp://localhost:8854/cam1')
videos = !ls {directory}/videos
for id, filename in enumerate(videos):
    stream(id, filename)
# stream(r'http://@192.168.0.39:8080/video')

##### Всякий хлам

In [14]:
import os

directory = '/home/pc/Downloads/dataset_from_clips_v2_new'
labels = !ls {directory}/labels
counter = 0
for label in labels:
    image = f'{directory}/images/{label.split(".")[0]}.png'
    if not os.path.exists(image):
        counter += 1
        os.remove(f'{directory}/labels/{label}')
counter

268

In [ ]:
import os

directory = '/home/pc/Downloads/project-21'
images = !ls {directory}/images
print(images)
counter = 0
for image in images:
    label = f'{directory}/labels/{image.split(".")[0]}.txt'
    if not os.path.exists(label):
        counter += 1
        os.remove(f'{directory}/images/{image}')
counter

In [4]:
import os
import cv2

directory = '/home/pc/Downloads/dataset_october/valid/images'
os.chdir(directory)

images = !ls 
counter = 0
for image in images:
    img = cv2.imread(image)
    img_gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
    cv2.imwrite(image, img_gray)
    counter += 1
counter

220

In [10]:
os.getcwd()

'/home/pc/Downloads'

##### Загрузка датасета

In [2]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("vladkemaev/dataset")

print("Path to dataset files:", path)

Resuming download from 217055232 bytes (3379498787 bytes left)...
Resuming download from https://www.kaggle.com/api/v1/datasets/download/vladkemaev/dataset?dataset_version_number=1 (217055232/3596554019) bytes left.


100%|██████████| 3.35G/3.35G [05:27<00:00, 10.3MB/s]

Extracting files...


Path to dataset files: /home/pc/.cache/kagglehub/datasets/vladkemaev/dataset/versions/1


##### Разбиение на train / test / valid

In [7]:
import os

directory = '/home/pc/Downloads/dataset_november'
train = f'{directory}/train'
valid = f'{directory}/valid'
test = f'{directory}/test'

counter = 0
images = !ls {train}/images
labels = !ls {train}/labels
for image, label in zip(images, labels):
    if counter % 5 == 0:
        # print(f'{valid}/images/{image}', '\n')
        os.replace(f'{train}/images/{image}', f'{valid}/images/{image}')
        os.replace(f'{train}/labels/{label}', f'{valid}/labels/{label}')
    if (counter - 1) % 10 == 0:
        # print(f'{test}/images/{image}', '\n')
        os.replace(f'{train}/images/{image}', f'{test}/images/{image}')
        os.replace(f'{train}/labels/{label}', f'{test}/labels/{label}')
    
    counter += 1
counter

1099

##### Выгрузка датасета

In [8]:
import kagglehub

# Download latest version
# path = kagglehub.dataset_download("dmitriymaslyaev/forklift-dataset-from-clips-v2")
kagglehub.dataset_upload('dmitriymaslyaev/dataset_november', '/home/pc/Downloads/datasets/dataset_november.zip')

Uploading Dataset https://www.kaggle.com/datasets/dmitriymaslyaev/dataset_november ...
Starting upload for file /home/pc/Downloads/datasets/dataset_november.zip


Uploading: 100%|██████████| 3.60G/3.60G [05:47<00:00, 10.3MB/s] 

Upload successful: /home/pc/Downloads/datasets/dataset_november.zip (3GB)


Your dataset instance has been created.
Files are being processed...
See at: https://www.kaggle.com/datasets/dmitriymaslyaev/dataset_november


### LXC контейнер инсталляция

In [ ]:
apt install htop
export CPU=amd
export nested_val=1
apt list --upgradable
nano /etc/apt/sources.list
apt install dirmngr ca-certificates software-properties-common apt-transport-https dkms curl -y
apt update
apt upgrade
curl -fSsL https://developer.download.nvidia.com/compute/cuda/repos/debian12/x86_64/3bf863cc.pub | sudo gpg --dearmor | sudo tee /usr/share/keyrings/nvidia-drivers.gpg > /dev/null 2>&1
echo 'deb [signed-by=/usr/share/keyrings/nvidia-drivers.gpg] https://developer.download.nvidia.com/compute/cuda/repos/debian12/x86_64/ /' | sudo tee /etc/apt/sources.list.d/nvidia-drivers.list
apt update
apt-cache policy cuda
apt install cuda=12.4.1-1
apt-cache policy cuda
apt-mark hold cuda
apt-mark showhold cuda
sudo apt-get install nvidia-container-toolkit
nvidia-ctk runtime configure --runtime=docker
chown root:video /dev/fb0
nvidia-smi
history 
chown root:video /dev/dri/card1
chown root:render /dev/dri/renderD128
docker context use default
systemctl restart docker
apt install nvidia-driver

In [ ]:
import os

files = os.listdir('/home/pc/frigate_test/qwe')
if files:
    files = [int(file.split('_')[0]) for file in files]
    files.sort()
    index = files[-1]+1
else:
    index = 1
index

1

In [48]:
import os

def get_index(destination:str=None):
    """
    Compute the index for file with estimated speed

    :destination: str - directory where videos store
    """
    try:
        files = os.listdir(destination)
        index = max(int(file.split('_')[0]) for file in files) + 1
    except:
        index = 1

    return index

get_index('/home/pc/frigate_test/qkkwe')

1